src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts

```diff
@@ -42,39 +45,48 @@ async function requestPriceForCurrency(chainId: number | undefined, currency: Cu
 
 // Fetches the INPUT and OUTPUT price and calculates initial Active rate
 // When return null it means we failed on price loading
+// TODO: rename it to useNativeBasedPrice
 export function useGetInitialPrice(): { price: Fraction | null; isLoading: boolean } {
   const { chainId } = useWeb3React()
   const { inputCurrency, outputCurrency } = useLimitOrdersTradeState()
-  const [isInputPriceLoading, setInputPriceLoading] = useState(false)
-  const [isOutputPriceLoading, setOutputPriceLoading] = useState(false)
   const [isLoading, setIsLoading] = useState(false)
-
-  const inputPrice = useAsyncMemo(() => {
-    setInputPriceLoading(true)
-    return requestPriceForCurrency(chainId, inputCurrency).finally(() => {
-      setInputPriceLoading(false)
-    })
-  }, [chainId, inputCurrency])
-
-  const outputPrice = useAsyncMemo(() => {
-    setOutputPriceLoading(true)
-    return requestPriceForCurrency(chainId, outputCurrency).finally(() => {
-      setOutputPriceLoading(false)
-    })
-  }, [chainId, outputCurrency])
-
-  const price = useMemo(() => {
-    if (!inputPrice || !outputPrice || inputPrice instanceof Error || outputPrice instanceof Error) {
-      return null
-    }
-
-    return new Fraction(inputPrice, outputPrice)
-  }, [outputPrice, inputPrice])
-
-  // To avoid loading state blinking
+  const [updateTimestamp, setUpdateTimestamp] = useState(Date.now())
+
+  const price = useAsyncMemo(
+    () => {
+      setIsLoading(true)
+
+      return Promise.all([
```

Comment:
would it make sense to create a function that would do this 2 request, and then then, so this part gets simplified? 

sth like 
```js
const price = useAsyncMemo(
    () => {
      setIsLoading(true)
      return getMarketPrice(chainId, inputCurrency outputCurrency)
              .finally(() => setLoading(false))
     })
```

Reply:
Thanks! Done